components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/js/idp_mgt_edit.js

```diff
@@ -1039,6 +1039,9 @@ jQuery('#advancedClaimMappingAddLink').click(function () {
     }
     advancedClaimMappinRowID++;
     $("#advanced_claim_id_count").val(advancedClaimMappinRowID + 1);
+
+    // Encoding the selectedIDPClaimName to avoid possible xss vulnerabilities.
+    selectedIDPClaimName = htmlEncode(selectedIDPClaimName);
```

Comment:
Should not this be encode for attribute.
htmlEncode may add incorrect div tag to the attribute.

```
value="' + selectedIDPClaimName + '"
```

Reply:
In here div tags are not added to the attribute. Here the text inside the dev tags are considered as text and and done the processing.
This is the recommended method in engineering guidelines [1] in section **2.7.1.3 JavaScript Specific Recommendations**

[1] - https://wso2.com/technical-reports/wso2-secure-engineering-guidelines

Reply:
+1